wgpu-core/src/command/render.rs

```diff
@@ -1279,9 +1279,7 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     }
                     RenderCommand::SetScissor(ref rect) => {
                         let scope = PassErrorScope::SetScissorRect;
-                        if rect.w == 0
-                            || rect.h == 0
-                            || rect.x + rect.w > info.extent.width
+                        if rect.x + rect.w > info.extent.width
```

Comment:
Unfortunately, this would make the Metal backend to fail. In order to remove this condition, we need to implement a small workaround in Metal backend to set the viewport to something dummy, as a replacement for zero scissor.